packages/mds-compliance-engine/engine/helpers.ts

```diff
@@ -0,0 +1,166 @@
+   * So we throw old events out and do not consider them.
+   * We also don't consider events that have no associated telemetry.
+   */
+  const filteredEvents = getEvents(events)
+  return { filteredEvents, deviceMap }
+}
+
+export function isPolicyActive(policy: Policy, end_time: number = now()): boolean {
+  if (policy.end_date === null) {
+    return end_time >= policy.start_date
+  }
+  return end_time >= policy.start_date && end_time <= policy.end_date
+}
+
+export function isRuleActive(rule: Rule): boolean {
+  if (!env.TIMEZONE) {
+    throw new RuntimeError('TIMEZONE environment variable must be declared!')
+  }
+
+  if (!moment.tz.names().includes(env.TIMEZONE)) {
+    throw new RuntimeError(`TIMEZONE environment variable ${env.TIMEZONE} is not a valid timezone!`)
+  }
+
+  const local_time = moment().tz(env.TIMEZONE)
+
+  if (!rule.days || rule.days.includes(Object.values(DAYS_OF_WEEK)[local_time.day()] as DAY_OF_WEEK)) {
+    if (!rule.start_time || local_time.isAfter(moment(rule.start_time, TIME_FORMAT))) {
+      if (!rule.end_time || local_time.isBefore(moment(rule.end_time, TIME_FORMAT))) {
+        return true
+      }
+    }
+  }
+  return false
+}
+
+export function isInVehicleTypes(rule: Rule, device: Device): boolean {
+  return !rule.vehicle_types || (rule.vehicle_types && rule.vehicle_types.includes(device.vehicle_type))
+}
+
+// Take a list of policies, and eliminate all those that have been superseded. Returns
+// policies that have not been superseded.
+export function getSupersedingPolicies(policies: Policy[]): Policy[] {
+  const prev_policies: string[] = policies.reduce((prev_policies_acc: string[], policy: Policy) => {
+    if (policy.prev_policies) {
+      prev_policies_acc.push(...policy.prev_policies)
+    }
+    return prev_policies_acc
+  }, [])
+  return policies.filter((policy: Policy) => {
+    return !prev_policies.includes(policy.policy_id)
+  })
+}
+
+export function getEvents(events: VehicleEvent[], end_time = now()): VehicleEvent[] {
+  return events
+    .filter((event: VehicleEvent) => {
+      /* Keep events that are less than two days old.
+       * This is a somewhat arbitrary window of time.
+       * Events with no telemetry are not used to process compliance.
+       */
+      return event.telemetry && event.timestamp > end_time - TWO_DAYS_IN_MS
+    })
+    .sort((e_1, e_2) => {
+      return e_1.timestamp - e_2.timestamp
+    })
+}
+
+export function createMatchedVehicleInformation(
+  device: Device,
+  event: VehicleEventWithTelemetry,
+  rule_applied_id?: UUID | null,
+  rules_matched?: UUID[]
+): MatchedVehicleInformation {
+  return {
+    device_id: device.device_id,
+    state: event.vehicle_state,
+    event_types: event.event_types,
+    timestamp: event.timestamp,
+    rules_matched: rules_matched || [],
+    rule_applied: rule_applied_id, // a device can only ever match at most one rule for the purpose of computing compliance
```

Comment:
Considering this comment is basically in the typedef already, I think it's unnecessary to dupe here as well. And if we add typedoc for the typedefs, then this should show up on intellisense I think?